### PR TITLE
update algorithms for perfectly forwarding Callable concept

### PR DESCRIPTION
**Pull request by ericniebler**
refs ericniebler/stl2#189

### CODE REVIEW

File: include/stl2/detail/algorithm/generate.hpp [hunk 1]
```diff
@@ -20,7 +20,9 @@
 // generate [alg.generate]
 //
 STL2_OPEN_NAMESPACE {
-	template <Callable F, OutputIterator<result_of_t<F&()>> O, Sentinel<O> S>
+	template <class F, OutputIterator<result_of_t<F&()>> O, Sentinel<O> S>
+	requires
+		models::Callable<F&>
```
Comment (CaseyCarter) [inline include/stl2/detail/algorithm/generate.hpp]: It's too late to require `Callable<F&>`, we've already instantiated `result_of_t<F&()>`. This needs to be:
```c++
template <class F, Iterator O, Sentinel<O> S>
requires
    models::Callable<F&> &&
    models::Writable<O, result_of_t<F&()>>
```

File: include/stl2/detail/algorithm/generate_n.hpp [hunk 2]
```diff
@@ -20,7 +20,9 @@
 // generate_n [alg.generate]
 //
 STL2_OPEN_NAMESPACE {
-	template <Callable F, OutputIterator<result_of_t<F&()>> O>
+	template <class F, OutputIterator<result_of_t<F&()>> O>
+	requires
+		models::Callable<F&>
```
Comment (CaseyCarter) [inline include/stl2/detail/algorithm/generate_n.hpp]: Same issue here as for `generate`.

**Comment by ericniebler** [comment]
Better?

**Comment by CaseyCarter** [comment]
Yip.

**Pull request merged**